lib/mongoid/criteria/queryable/extensions/time.rb

```diff
@@ -28,7 +28,7 @@ def __evolve_date__
           #
           # @since 1.0.0
           def __evolve_time__
```

Comment:
It's interesting that this is called "evolve", which to me implies that the mutation might have originally been intention (although I might just not be realizing that this is an override of something else). That being said, I definitely agree that we shouldn't be mutating the arguments users pass in to queries, so I'm fine with this change.